File: test-suite/src/test/java/io/micronaut/kafka/docs/consumer/config/ConfigProductListenerTest.java
package io.micronaut.kafka.docs.consumer.config;

import io.micronaut.context.ApplicationContext;
import io.micronaut.kafka.docs.Product;
import io.micronaut.testcontainers.kafka.Kafka;
import org.junit.jupiter.api.Test;

import java.util.HashMap;
import java.util.Map;

import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;

class ConfigProductListenerTest {

    @Test
    void testSendProduct() {
        Map<String, String> kafkaProps = Kafka.getProperties();

        Map<String, Object> config = new HashMap<>(kafkaProps);
        config.put("kafka.enabled", "true");
        config.put("spec.name", "ConfigProductListenerTest");

        try (ApplicationContext ctx = ApplicationContext.run(config)) {
            assertDoesNotThrow(() -> {
                Product product = new Product("Blue Trainers", 5);
                ProductClient client = ctx.getBean(ProductClient.class);
                client.send("Nike", product);
            });
        }
    }
}

File: test-suite-kotlin/src/test/kotlin/io/micronaut/kafka/docs/consumer/config/ConfigProductListenerTest.kt
package io.micronaut.kafka.docs.consumer.config

import io.micronaut.context.ApplicationContext
import io.micronaut.core.util.StringUtils
import io.micronaut.kafka.docs.Product
import org.junit.jupiter.api.Assertions
import org.junit.jupiter.api.Test
import java.util.Map

class ConfigProductListenerTest {

    @Test
    fun testSendProduct() {
        ApplicationContext.run(
            mapOf("kafka.enabled" to StringUtils.TRUE, "spec.name" to "ConfigProductListenerTest")
        ).use { ctx ->
            Assertions.assertDoesNotThrow {
                val product = Product("Blue Trainers", 5)
                val client = ctx.getBean(ProductClient::class.java)
                client.send("Nike", product)
            }
        }
    }
}
